File: leetcode_0349/java/leetcode_0349.java
import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

public class leetcode_0349 {
    static class Solution {
        public static int[] intersection(int[] nums1, int[] nums2) {
            Set<Integer> record = new HashSet();
            Set<Integer> inter = new HashSet();
            for(int num : nums1)
                record.add(num);
            for(int num : nums2){
                if(record.contains(num))
                    inter.add(num);
            }
            int[] res = new int[inter.size()];
            int i = 0;
            for(Integer tmp:inter)
                res[i++] = tmp.intValue();
            return res;
        }
    }
    public static void main(String[] args) {
        int[] nums1 = {1,2,2,1};
        int[] nums2 = {2,2};
        int[] res = Solution.intersection(nums1, nums2);
        System.out.println(Arrays.toString(res));
        //result
        //[2]
    }
}

File: leetcode_0349/cpp/leetcode_0349.cpp
//
// Created by wangzy on 2022/3/6.
//
#include <iostream>
#include <vector>
#include <unordered_set>
#include <sstream>

using namespace std;

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> record;
        unordered_set<int> inter;
        for(auto num : nums1)
            record.emplace(num);
        for(auto num : nums2){
            if(record.count(num))
                inter.emplace(num);
        }
        return vector<int>(inter.begin(),inter.end());
    }
};

template<typename T>
void printVec(std::vector<T> l){
    std::string res = "[";
    std::string tmp = "";
    std::stringstream ss;
    for(auto ele : l){
        ss << ele;
        ss >> tmp;
        res += tmp + ",";
        ss.clear();
    }
    res = res.substr(0,res.size()-1);
    res += "]";
    std::cout<<res<<std::endl;
}

int main(int argc,char* argv[]){
    vector<int> nums1 = {1,2,2,1};
    vector<int> nums2 = {2,2};

    vector<int> res = Solution().intersection(nums1,nums2);
    printVec(res);

    //[2]
}
